apps/crawler/crawler/s3/crawler/s3_crawler.py

```diff
@@ -4,21 +4,31 @@
 
 import boto3
 import botocore.client
+from botocore import UNSIGNED
+from botocore.config import Config
+from botocore.exceptions import NoCredentialsError
 import sys
 
 
+def usage(val: int) -> None:
+    print("Usage: poetry run python s3_crawler.py bucket_name prefix_value")
+    sys.exit(val)
+
+
 class S3Crawler:
     def __init__(
         self,
         bucket_location: str,
         prefix: str,
+        anon: bool,
         boto_session_args: list[Any] = [],
         boto_session_kwargs: dict[str, Any] = {},
     ):
         self._bucket_location = bucket_location
         self._prefix = prefix
+        self._anon = anon
```

Comment:
Auto detect that we want anon mode if AWS_ACCESS_KEY_ID AWS_SECRET_ACCESS_KEY are empty.

Reply:
It's not that simple.  Boto3 will look in lots of filesystem places for credentials.